op-chain-ops/cmd/check-derivation/main.go

```diff
@@ -0,0 +1,432 @@
+		}
+		txData = &types.DynamicFeeTx{
+			ChainID:   chainId,
+			Nonce:     nonce,
+			GasTipCap: gasTipCap,
+			GasFeeCap: gasPrice,
+			Gas:       gasLimit,
+			To:        &randomAddress,
+			Value:     amount,
+			Data:      data,
+		}
+	default:
+		return nil, fmt.Errorf("unsupported tx type: %d", txType)
+	}
+
+	tx := types.NewTx(txData)
+
+	signer := types.NewLondonSigner(chainId)
+	if !protected {
+		if txType == types.LegacyTxType {
+			signer = types.HomesteadSigner{}
+		} else {
+			return nil, errors.New("typed tx cannot be unprotected")
+		}
+	}
+
+	signedTx, err := types.SignTx(tx, signer, privateKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign transaction: %w", err)
+	}
+	return signedTx, nil
+}
+
+// confirmTransaction polls receipts to confirm transaction is included in the block.
+func confirmTransaction(ctx context.Context, ethClient *ethclient.Client, l2BlockTime uint64, txHash common.Hash) (eth.BlockID, error) {
+	var retryCount uint64
+	for {
+		receipt, _ := ethClient.TransactionReceipt(ctx, txHash)
+		if retryCount > 30 {
+			return eth.BlockID{}, fmt.Errorf("transaction confirmation failure: txHash: %s", txHash.String())
+		}
+		if receipt == nil {
+			log.Info("Waiting for transaction receipt", "txHash", txHash.String())
+			retryCount++
+			// wait at least l2 block time
+			time.Sleep(time.Duration(l2BlockTime) * time.Second)
+			continue
+		}
+		block := eth.BlockID{
+			Hash:   receipt.BlockHash,
+			Number: receipt.BlockNumber.Uint64(),
+		}
+		log.Info("Transaction receipt found", "block", block, "status", receipt.Status)
+		return block, nil
+	}
+}
+
+// checkConsolidation sends transactions and ensures them to be included in unsafe block.
+// Then polls safe head to check unsafe blocks which includes sent tx are consolidated.
+func checkConsolidation(cliCtx *cli.Context) error {
+	ctx := context.Background()
+	clients, err := newClientsFromContext(ctx, cliCtx)
+	if err != nil {
+		return err
+	}
+	var pollingInterval = time.Duration(cliCtx.Int("polling-interval")) * time.Millisecond
+	privateKey, err := getPrivateKey(cliCtx)
+	if err != nil {
+		return err
+	}
+	from, err := getSenderAddress(privateKey)
+	if err != nil {
+		return err
+	}
+	txCount := cliCtx.Int("tx-count")
+	if txCount < 4 {
+		return fmt.Errorf("tx count %d is too low. requires minimum 4 txs to test all tx types", txCount)
+	}
+	l2ChainID := new(big.Int).SetUint64(cliCtx.Uint64("l2-chain-id"))
+	l2BlockTime := uint64(2)
+	rollupCfg, err := rollup.LoadOPStackRollupConfig(l2ChainID.Uint64())
+	if err == nil {
+		l2BlockTime = rollupCfg.BlockTime
+	} else {
+		log.Warn("Superchain config not loaded", "l2-chain-id", l2ChainID)
+		log.Warn("Using default config", "l2-block-time", l2BlockTime)
+	}
+	rng := rand.New(rand.NewSource(1337))
+	// txMap maps txHash to blockID
+	txMap := make(map[common.Hash]eth.BlockID)
+	for i := 0; i < txCount; i++ {
+		var tx *types.Transaction
+		switch i % 4 {
+		case 0:
```

Comment:
you could dedupe these long lines of code by assigning variables to the type and the bool (the 2 args that are different) and then call `getRandomSignedTransaction` once below